cloud/previews/previews.go

```diff
@@ -176,3 +215,55 @@ func (o CreatePreviewOpts) validate() error {
 
 	return errs.AsError()
 }
+
+func (o CreateStackPreviewOpts) validate() error {
+	errs := errors.L()
+	if o.StackPreviewID == "" {
+		errs.Append(errors.E("stack preview id is empty"))
+	}
+
+	if string(o.OrgUUID) == "" {
+		errs.Append(errors.E("org uuid is empty"))
+	}
+
+	if o.Status == "" {
+		errs.Append(errors.E("status is empty"))
+	}
+
+	if o.ChangesetDetails != nil {
+		if o.ChangesetDetails.Provisioner == "" {
+			errs.Append(errors.E("provisioner is empty"))
+		}
+		if o.ChangesetDetails.ChangesetASCII == "" {
+			errs.Append(errors.E("changeset ascii is empty"))
+		}
+		if o.ChangesetDetails.ChangesetJSON == "" {
+			errs.Append(errors.E("changeset json is empty"))
+		}
+	}
+
+	return errs.AsError()
+}
+
+// DerivePreviewStatus derives the preview status from the exit code, run error
+// and whether the run is a terraform plan
+func DerivePreviewStatus(exitCode int, isTerraformPlan bool) StackPreviewStatus {
+	if isTerraformPlan {
+		switch exitCode {
+		case 2:
+			return PreviewStatusChanged
+		case 0:
+			return PreviewStatusUnchanged
+		}
+	}
+
+	switch exitCode {
+	case -1:
+		return PreviewStatusCanceled
+	case 0:
+		return PreviewStatusUnchanged
```

Comment:
Why not `Changed`?

Reply:
@sixstone-qq I assumed that the `Changed` status is only used when there are changes in the terraform plan. 
The `unchanged` status for non-terraformplan commands returning exit code 0 does seem a bit strange though.
I am wondering if it would make sense to add an `ok` status for these cases. What do you think?

Reply:
For me, it should assume `changed` even though it's unknown @mariux do you want to chime in?